cpp/arcticdb/version/symbol_list.cpp

```diff
@@ -159,25 +163,34 @@ bool SymbolList::can_update_symbol_list(const std::shared_ptr<Store>& store,
             );
             list_segment.end_row();
         }
-        if(symbols.empty()) {
-            google::protobuf::Any any = {};
-            arcticdb::proto::descriptors::SymbolListDescriptor metadata;
+
+        arcticdb::proto::descriptors::SymbolListDescriptor metadata;
+        if (existing_descriptor) {
+            metadata.CopyFrom(*existing_descriptor);
+            if (ARCTICDB_VERSION_INT < metadata.min_writer_version()) {
+                metadata.set_min_writer_version(ARCTICDB_VERSION_INT);
+            }
+        } else {
             metadata.set_enabled(true);
-            any.PackFrom(metadata);
-            list_segment.set_metadata(std::move(any));
+            util::set_static_writer_info_fields(*metadata.mutable_initial_writer());
```

Comment:
I don't really think that writing telemetry data into the compacted symbol list is going to help us debug anything, because compaction, although it is somewhat complex compared to the other operations, represents only 0.1% of the symbol lists actions, so knowing the version that did the compaction is likely to be a red herring in terms of diagnosing symbol list corruption.